### PR TITLE
Extended fort defence coverage

### PR DESCRIPTION
**Pull request by valdisz**
I have implemented an extended fort (all fortifications) defense feature.

This feature extends fort defense bonus on garrisons joining battles in adjacent regions. Bonus will be applied only if the garrison is defending, i.e. treat this as a siege on the fort. Ships do not have such a feature it applies only to buildings.

Opening new PR as the old one (#67) was closed. **This is finished feature.**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,22 +1,16 @@
 .DS_Store
 .vscode/
+.vs/
 *.code-workspace
-obj
+obj/
 basic/basic
-standard/obj
```
Comment (artyomtrityak) [inline .gitignore:6]: why did you remove this from .gitignore?

File: battle.cpp [hunk 2]
```diff
@@ -686,6 +713,12 @@ void Game::GetSides(ARegion *r, AList &afacs, AList &dfacs, AList &atts,
 				}
 
 				if (add == ADD_ATTACK) {
+					if (extendFortDefence && !fortDefenceCleared) {
+						// EXTENDED_FORT_DEFENCE_COVERAGE is not working for attacker garrisons
```
Comment (artyomtrityak) [inline battle.cpp]: I think we should keep same behavior for attack and defense so let's remove this check and add fort defenses for attacking too.

File: battle.cpp [hunk 3]
```diff
@@ -592,18 +609,30 @@ void Game::GetSides(ARegion *r, AList &afacs, AList &dfacs, AList &atts,
 	int noaida = 0, noaidd = 0;
 	for (int i=-1;i<j;i++) {
 		ARegion * r2 = r;
-		if (i>=0) {
+		bool fromNeighboar = i >= 0;
```
Comment (artyomtrityak) [inline battle.cpp]: fromNeighbor

File: .gitignore [hunk 4]
```diff
@@ -1,22 +1,16 @@
...
-havilah/obj
 havilah/havilah
 havilah/html/havilah.html
-neworigins/obj
 neworigins/neworigins
 neworigins/html/neworigins.html
-neworigins_v2/obj
 neworigins_v2/neworigins_v2
-neworigins_v2/html/neworigins.html
+neworigins_v2/html/neworigins.html
```
Comment (artyomtrityak) [inline .gitignore]: please return new line

File: battle.cpp [hunk 5]
```diff
@@ -592,18 +609,30 @@ void Game::GetSides(ARegion *r, AList &afacs, AList &dfacs, AList &atts,
...
-			}
-		} else {
-			forlist(&r2->objects) {
-				Object * o = (Object *) elem;
-				/* Set building capacity */
+		}
+
+		forlist (&r2->objects) {
+			Object * o = (Object *) elem;
+			bool extendFortDefence =
```
Comment (artyomtrityak) [inline battle.cpp]: this logic seems can be improved. Can you create if else statement with such condition

```
if (Globals->EXTENDED_FORT_DEFENCE_COVERAGE && o->IsBuilding() && !o->IsFleet()) {
  ... add bonus ...
} else {
   ... do not add bonus ...
}
```

so you do not need to clear anything and bonus is added only if it passes all tests.

File: battle.cpp (hunk 2 above)
Comment (valdisz) [reply battle.cpp]: Makes sense.

File: battle.cpp (hunk 3 above)
Comment (valdisz) [reply battle.cpp]: 🤐

File: battle.cpp (hunk 5 above)
Comment (valdisz) [reply battle.cpp]: I'll look at what I can do

File: .gitignore (hunk 1 above)
Comment (valdisz) [reply .gitignore:6]: `obj/` replaces all removed folder specific `whatever/obj` rules

**Comment by valdisz** [comment]
@artyomtrityak this should be ready

**Comment by artyomtrityak** [comment]
closing in favor for https://github.com/artyomtrityak/Atlantis/pull/96/commits/85682557eb5bfaef47a53df3db8d167ea4939dde

this commit used this PR as a reference.
